ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayDimensions/DisplayDimensions.cpp

```diff
@@ -0,0 +1,190 @@
+  m_mmpk = new MobileMapPackage(mapPackagePath, this);
+
+  // Make connections between the mmpk's doneLoading and errorOccurred signal and the addMapToMapView and handleError methods respectively.
+  connect(m_mmpk, &MobileMapPackage::doneLoading, this, &DisplayDimensions::addMapToMapView);
+  connect(m_mmpk, &MobileMapPackage::errorOccurred, this, &DisplayDimensions::handleError);
+
+  // Load the mmpk.
+  m_mmpk->load();
+}
+
+DisplayDimensions::~DisplayDimensions() = default;
+
+void DisplayDimensions::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DisplayDimensions>("Esri.Samples", 1, 0, "DisplayDimensionsSample");
+}
+
+void DisplayDimensions::addMapToMapView(const Error& error)
+{
+  // If no error's have occurred, the mmpk is loaded, and there is only one map in the mmpk, the map in the mmpk can be
+  // assigned to the MapView and the viewpoint set. Any errors will be managed by the handleError method.
+  if (error.isEmpty() && m_mmpk->loadStatus() == LoadStatus::Loaded && m_mmpk->maps().count() > 0)
+  {
+    // Enable the checkboxes.
+    setCheckBoxesEnabled(true);
+
+    // Assign the map in the mmpk to m_map.
+    m_map = m_mmpk->maps().at(0);
+    // Set the viewpoint.
+    m_map->setInitialViewpoint(Viewpoint(Point(-3.3098678, 55.9074044, SpatialReference::wgs84()), 30000));
+    // Set the minimum scale to prevent zooming out too far.
+    m_map->setMinScale(35000);
+    // Set m_map as the MapView's map.
+    m_mapView->setMap(m_map);
+
+    // From the map's layers, find the Dimension Layer.
+    findDimensionLayer();
+  }
+  else
+    // If the map hasn't loaded or an error has occurred, disable the checkboxes in the UI.
+    setCheckBoxesEnabled(false);
+}
+
+void DisplayDimensions::findDimensionLayer()
+{
+  LayerListModel* layers = m_mapView->map()->operationalLayers();
+
+  for (Layer* layer : *layers)
+  {
+    if (layer->layerType() == LayerType::DimensionLayer)
+    {
+      // The current layer, which is the DimensionLayer, has type Layer*. Ths needs to be converted to a DimensionLayer*.
+      m_dimensionLayer = dynamic_cast<DimensionLayer*>(layer);
+      // Use the name of the Dimension Layer to define the title of the UI element.
+      setDimensionLayerName(m_dimensionLayer->name());
+      // There is only one Dimension Layer, so we can break out of the loop.
+      break;
+    }
+  }
+}
+
+void DisplayDimensions::handleError(const Error& error)
+{
+  if (error.additionalMessage().isEmpty())
+    setErrorMessage(error.message());
+  else
+    setErrorMessage(error.message() + "\n" + error.additionalMessage());
+}
+
+MapQuickView* DisplayDimensions::mapView() const
+{
+  return m_mapView;
+}
+
+// Set the view (created in QML)
+void DisplayDimensions::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
```

Comment:
good chance `m_map` hasn't been set by the time this is called since we need to load it from the mmpk